app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java

```diff
@@ -247,9 +229,17 @@ public void onClick(DialogInterface dialog, int id) {
 
     public void createAccount(View v){
         //TODO:: send intent to register activity
-        Intent registerUser = new Intent(v.getContext(), RegisterActivity.class);
-        startActivity(registerUser);
-        finish();
+        ConnectivityManager cm = (ConnectivityManager) getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        boolean isConnected = activeNetwork != null && activeNetwork.isConnectedOrConnecting();
+
+        if (isConnected) {
+            Intent registerUser = new Intent(v.getContext(), RegisterActivity.class);
+            startActivity(registerUser);
+            finish();
+        } else {
+            Toast.makeText(this, "Connect to WiFi or data then please try again :)", Toast.LENGTH_SHORT).show();
```

Comment:
Store this in strings.xml to be consistent with how we're handling error toasts elsewhere.

Reply:
Agreed.